### ISSUE TITLE
GEM daqmonitor not visualizable in the web interface

### ISSUE THREAD

**Issue by cgalloni** [cms-gem-daq-project/cmsgemos#254]
Gemdaqmonitor package  from branch ` `cmsgemos/develop` correctly compiled, but  the web interface visualization fails. 

## Brief summary of issue

Gemdaqmonitor package correctly compiled, but  the web interface visualization fails, giving error about undefined symbol 
```shell
/opt/xdaq/bin/xdaq.exe: symbol lookup error: //afs/cern.ch/work/c/cgalloni/cms-gem-daq-proj/cmsgemos/gemdaqmonitor/lib/linux/x86_64_centos7/libgemdaqmonitor.so: undefined symbol: _ZN11gemhardware14getPackageInfoEv
```
```shell
nm lib/linux/x86_64_centos7/libgemdaqmonitor.so |c++filt|grep gemhardware
                 U gemhardware::getPackageInfo()
```

### Types of issue

- [x] Bug report (report an issue with the code)
- [ ] Feature request (request for change which adds functionality)

## Expected Behavior

The package should be visualized in the web interface

### Steps to Reproduce (for bugs)

1.Setup your building env
2. Run compilation
```shell
cd $BUILD_HOME/cmsgemos/gemdaqmonitor
make clean && make all -j 8
```
3. Visualize: ```shell /opt/xdaq/bin/xdaq.exe -p20500 -c $PWD/xml/gemmonitor.xml```
4.

## Your Environment

* Version used:  branch `cmsgemos/develop`
* Shell used: bash

**Comment by jsturdy** [comment]
TL;DR: there is an error in the `gemdaqmonitor.xml` file, `libgemhardware_managers.so` should be loaded into the process.

### Long version:
This is due to one or two issues
The first is that `gemdaqmonitor` is built with a dependency on `gemhardware`, but the xml startup config does not load the appropriate shared object.
It seems that the only dependency that `gemdaqmonitor` has is on some conversion functions, which *should* be placed in the `gemutils` package, but for legacy reasons are currently in `gemhardware`.
They are built into both `libgemhardware_managers.so` and `libgemhardware_devices.so`, so you should be able to add either one of these to your process startup and the issue should be resolved.
Actually, you will have to load `libgemhardware_managers.so` due to the second part of this issue.

The second part of the issue stems from the way `xdaq` does dynamic checking of library compatibility at runtime.
This is controlled in the `version.h`/`version.cc` files that are built into each shared object.
For the case of `gemhardware` (which is now listed as a dependency in the version files for `gemdaqmonitor`), I had not been building the version info into the devices library, instead, only building it into the managers library.
I was trying to reduce as much as possible the dependencies on the devices library, so it could easily be distributed without many external dependencies.  I ultimately was not able to strip this down completely, so a couple of things could be done:
* Create separate dependencies for each library
  * The way the `xdaq` structure likes this would mean a rearrangement of the `gemhardware` package from:
```
gemhardware/
  src
    common
      version.h
  include
    gem
      hw
        version.h
```
to:
```
gemhardware/
  devices
    src
      common
        version.h
    include
      gem
        hw
          devices
            version.h
  managers
    src
      common
        version.h
    include
      gem
        hw
          managers
            version.h
```

I have tried to make a minimal change to:
```
gemhardware
  src
    common
      devices
        version.h
      managers
        version.h
  include
    gem
      hw
        devices
          version.h
        managers
          version.h
```
and built into each library the version info in a non-standard (xdaq) way.
It seems to work,  was able to start the `DaqMonitor`, though I did not test any more deeply

If the minimal way doesn't work, a refactor of the package into the "proper" way will have to be done (which is probably a good thing in any event).

But the even better way would be to migrate the dependency of `gemdaqmonitor` on `gemhardware` to the `gemutils` function (again, it was because I was trying for minimal extra dependencies on `gemhardware_devices` that this exits)

**Comment by mexanick** [comment]
The issue is resolved in the upstream. Closing
